src/Server/KeeperTCPHandler.cpp

```diff
@@ -415,6 +415,12 @@ void KeeperTCPHandler::runImpl()
             log_long_operation("Polling socket");
             if (result.has_requests && !close_received)
             {
+                if (in->eof())
```

Comment:
How is it possible if epoll returned ready for requests socket or `in->available()` return non zero (`result.has_requests`)?

Reply:
epoll by default monitors for EPOLLHUP (so as poll via PULLHUP)
This can be rewritten to EPOLLHUP/EPOLLRDHUP/POLLHUP, but this will be more code.